tes-adapter/src/main/java/com/epam/pipeline/tesadapter/service/TaskMapper.java

```diff
@@ -78,18 +80,30 @@ private TesExecutor getExecutorFromTesExecutorsList(List<TesExecutor> tesExecuto
     public TesTask mapToTesTask(PipelineRun run) {
         return TesTask.builder()
                 .id(String.valueOf(run.getId()))
-                .state(createTesState(run.getStatus()))
                 .name(run.getPodId())
                 .resources(createTesResources(run))
                 .executors(createListExecutor(run))
                 .inputs(createTesInput(ListUtils.emptyIfNull(run.getPipelineRunParameters())))
                 .outputs(createTesOutput(ListUtils.emptyIfNull(run.getPipelineRunParameters())))
                 .creationTime(run.getStartDate().toString())
+                .state(createTesState(run))
                 .build();
     }
 
-    private TesState createTesState(TaskStatus status) {
-        switch (status) {
+    private TesState createTesState(PipelineRun run) {
+        List<PipelineTask> pipelineTaskList = cloudPipelineAPIClient.loadPipelineTasks(run.getId());
+        for (PipelineTask p : pipelineTaskList) {
```

Comment:
would it work for status RUNNING.
f.e.:
you have pipeline with status RUNNING and tasks like:
`Console`
`InputData`
`Install...`
`InitializeEnvironment`

in this case you have `InitializeEnvironment` but your code will return by condition `else if ((p.getName().equalsIgnoreCase("Console") &&
                    !p.getName().equalsIgnoreCase("InitializeEnvironment"))
                    && pipelineTaskList.size() > 1)` is it right?

Also I think that it could be wrote more accurate with stream API